src/row_codec.rs

```diff
@@ -0,0 +1,272 @@
+use crate::db_state::RowAttribute;
+use crate::error::SlateDBError;
+use crate::types::{KeyValueDeletable, ValueDeletable};
+use bitflags::bitflags;
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+
+bitflags! {
+    pub(crate) struct RowFlags: u8 {
+        const Tombstone = 0x01;
```

Comment:
I think it would be easier to grok if we use binary representation here - e.g. `0b00001`. Then additional flags would use values like `0b00010`, `0b00100`, `0b01000`, `0b10000` instead of `0x2`, `0x4`, `0x8`, `0x10`